### ISSUE TITLE
Training dataset larger than 2GB

### ISSUE THREAD

**Issue by paycade** [tensorflow/tfjs#7015]
Trying to train a dataset larger than 2GB provides the following error.

`RangeError [ERR_FS_FILE_TOO_LARGE]: File size (5209717379) is greater than 2 GiB
    at new NodeError (node:internal/errors:387:5)
    at tryCreateBuffer (node:fs:433:13)
    at Object.readFileSync (node:fs:478:14)
    at FileDataSource.<anonymous> (/Users/jose/stocksapp/MachineLearningConsole/node_modules/@tensorflow/tfjs-data/dist/tf-data.node.js:4965:37)
    at step (/Users/jose/stocksapp/MachineLearningConsole/node_modules/@tensorflow/tfjs-data/dist/tf-data.node.js:147:27)
    at Object.next (/Users/jose/stocksapp/MachineLearningConsole/node_modules/@tensorflow/tfjs-data/dist/tf-data.node.js:96:53)
    at /Users/jose/stocksapp/MachineLearningConsole/node_modules/@tensorflow/tfjs-data/dist/tf-data.node.js:89:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/jose/stocksapp/MachineLearningConsole/node_modules/@tensorflow/tfjs-data/dist/tf-data.node.js:75:12)
    at FileDataSource.iterator (/Users/jose/stocksapp/MachineLearningConsole/node_modules/@tensorflow/tfjs-data/dist/tf-data.node.js:4960:16) {
  code: 'ERR_FS_FILE_TOO_LARGE'`

This is using const { input } = require("@tensorflow/tfjs-node-gpu");

When calling method.

`  const trainingData = tf.data.csv("file://"+dirname+"/processed/sp500.csv", {
    columnConfigs : 
     { label : { isLabel : true}
     }
                              })`

**Comment by rthadur** [comment]
I guess there is upper limit to read file in nodejs , @pyu10055 could you please confirm ?

**Comment by pyu10055** [comment]
@paycade readFileSync has a 2G size limit, to allow files above 2GB, tfjs-data will need to use the stream file api, contributions are welcomed.
